### PR TITLE
x86/64: Add support for AMD SOC Ethernet controller

### PR DESCRIPTION
**Pull request by fishbaoz**
Signed-off-by: Zheng Bao <fishbaozi@gmail.com>

### CODE REVIEW

**Review by fishbaoz — COMMENTED** [review]
It is tested on v1000.

**Comment by nasbdh9** [comment]
also, please update the submission information

**Comment by fishbaoz** [comment]
> also, please update the submission information

Added

**Comment by Borromini** [comment]
@fishbaoz I believe the prefix for your PR title should be 'x86/64:', not 'amd_64'. Use the target/subtarget combination OpenWrt uses, Debian uses 'amd64'.

Also add your tested comment *in* your commit message please, that's helpful info that should go in the commit message.

File: package/kernel/linux/modules/netdevices.mk [hunk 1]
```diff
@@ -716,13 +716,30 @@ define KernelPackage/ixgbevf
...
  Kernel modules for Intel(R) 82599 Virtual Function Ethernet adapters.
 endef
 
 $(eval $(call KernelPackage,ixgbevf))
 
 
+define KernelPackage/amd-xgbe
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=AMD Ethernet support
+  DEPENDS:=@PCI_SUPPORT +kmod-lib-crc32c +kmod-ptp +kmod-libphy
```
Comment (Borromini) [inline package/kernel/linux/modules/netdevices.mk:730]: I'd have this depend on x86, or even x86_64 if this is only for x86_64 systems.

Like this:

`DEPENDS:=@TARGET_x86 @PCI_SUPPORT +kmod-lib-crc32c +kmod-ptp +kmod-libphy`

or `@TARGET_x86_64` if this only applies to 64 bit systems.

**Review by Borromini — CHANGES_REQUESTED** [review]

File: package/kernel/linux/modules/netdevices.mk [hunk 2]
```diff
@@ -688,7 +688,7 @@ $(eval $(call KernelPackage,igbvf))
 define KernelPackage/ixgbe
   SUBMENU:=$(NETWORK_DEVICES_MENU)
   TITLE:=Intel(R) 82598/82599 PCI-Express 10 Gigabit Ethernet support
-  DEPENDS:=@PCI_SUPPORT +kmod-mdio +kmod-ptp +kmod-hwmon-core +kmod-libphy +LINUX_5_10:kmod-mdio-devres
+  DEPENDS:=@PCI_SUPPORT @TARGET_x86 +kmod-mdio +kmod-ptp +kmod-hwmon-core +kmod-libphy +LINUX_5_10:kmod-mdio-devres
```
Comment (Borromini) [inline package/kernel/linux/modules/netdevices.mk:691]: This would be something for a separate patch I think, since this doesn't affect the AMD network controller you're adding support for in this PR.

**Review by Borromini — CHANGES_REQUESTED** [review]

File: package/kernel/linux/modules/netdevices.mk (hunk 2 above)
Comment (fishbaoz) [reply package/kernel/linux/modules/netdevices.mk]: done.

**Comment by fishbaoz** [comment]
done

**Comment by fishbaoz** [comment]
Moved to target/linux/x86/modules.mk

**Comment by hauke** [comment]
Thank you for the patch. I applied it to master.
